pkg/visor/rpc_client_serve.go

```diff
@@ -25,12 +29,29 @@ func isDone(ctx context.Context) bool {
 func ServeRPCClient(ctx context.Context, log logrus.FieldLogger, dmsgC *dmsg.Client, rpcS *rpc.Server, rAddr dmsg.Addr, errCh chan<- error) {
 	const maxBackoff = time.Second * 5
 	retry := netutil.NewRetrier(log, netutil.DefaultInitBackoff, maxBackoff, netutil.DefaultTries, netutil.DefaultFactor)
-
+	pubkey := cipher.PubKey{}
 	for {
 		var conn net.Conn
 		err := retry.Do(ctx, func() (rErr error) {
 			log.Info("Dialing...")
 			addr := dmsg.Addr{PK: rAddr.PK, Port: rAddr.Port}
+			if skyenv.AutoPeer {
+				var hvkey string
+				hvkey, err := script.Exec(skyenv.AutoPeercmd).String()
+				if err != nil {
+					log.Error("error autopeering")
+				} else {
+					hvkey = strings.TrimSuffix(hvkey, "\n")
+					hypervisorPKsSlice := strings.Split(hvkey, ",")
+					for _, pubkeyString := range hypervisorPKsSlice {
```

Comment:
I dont quite understand why you are trying to put all the auto peering functionality inside of the source code and then execute the peerCommand via this scripting library.

Reply:
because it was executing it essentially the same way that I would test it.

I've revised this PR to remove the use of bitfield/script